### PR TITLE
Hm306 constants ci

### PR DESCRIPTION
**Pull request by aschnapp**
### Overview
This moves all constant generation into CI
It does not change backend constants as the generated one still needs to be integrated. This pr does add a comment on how to use the auto generated backend constants.

### Includes:
Please see files for changes

### Developer Checklist:
- [x] The code follows the Quality Policy file on Google Drive
- [x] My code has been developer-tested and includes unit tests
- [x] I have considered proper use of exceptions
- [x] I have eliminated IDE warnings

### Notes:
This PR does not change anything to deployment. When deploying the client (whoever will work on that) you will need to generate constants before it can run correctly. 

### Refs:
- [HM-306](https://the-halo-mod.atlassian.net/jira/software/projects/HM/boards/1?selectedIssue=HM-306)

### CODE REVIEW

File: server/run.sh [hunk 1]
```diff
@@ -17,5 +17,7 @@ do
       export FLASK_APP=halomod_app
       export FLASK_ENV=development
       flask run;;
+    "--generate-constants" | "-gc" )
```
Comment (aneuhold) [inline server/run.sh:20]: Beautiful shortcut 😄

**Review by aneuhold — APPROVED** [review]
Awesome. Nice implementation. Works pretty cleanly it looks like.

**Comment by aneuhold** [comment]
#89 seems relevant to this PR. It would be helpful to Steven to have some documentation on how the client CI works and what it is doing in the file perhaps. Or in the main README documentation.

**Comment by aschnapp** [comment]
> #89 seems relevant to this PR. It would be helpful to Steven to have some documentation on how the client CI works and what it is doing in the file perhaps. Or in the main README documentation.

Let's add this as another User story, we dont have any documentation for that yet and it would be better for a task to create one from scratch.

**Review by vperuzzi — APPROVED** [review]

**Pull request merged**